pumpkin-lib/src/engine/constraint_satisfaction_solver.rs

```diff
@@ -1426,9 +1466,127 @@ impl ConstraintSatisfactionSolver {
             ),
             "Checking the propagations performed by the propagator led to inconsistencies!"
         );
+
         result
     }
 
+    /// Introduces any propagations to the proof by introducing them as nogoods.
```

Comment:
This should be phrased more clearly, it now appears that you're introducing _any_ propagations but you're only introducing the propagations which occurred at the root level